src/tess_asteroids/movingtpf.py

```diff
@@ -669,25 +734,72 @@ def _save_tpf(self, file_name: Optional[str] = None, save_loc: str = ""):
                 ]
             ),
         )
-        table_hdu.header["EXTNAME"] = "PIXELS"
+        table_hdu_spoc.header["EXTNAME"] = "PIXELS"
 
-        # Create aperture HDU
-        # >>>>> UPDATE IN FUTURE WITH REAL APERTURE <<<<<
-        aperture_hdu = fits.ImageHDU(
-            data=np.ones(self.shape),
+        # Create HDU containing average aperture
+        # Aperture has values 0 and 2, where 0/2 indicates the pixel is outside/inside the aperture.
+        # This format is used to be consistent with the aperture HDU from SPOC.
+        aperture_hdu_average = fits.ImageHDU(
+            data=np.nanmedian(self.aperture_mask, axis=0).astype("int32") * 2,
             header=fits.Header(
                 [*self.cube.output_secondary_header.cards, *wcs_header.cards]
             ),
         )
-        aperture_hdu.header["EXTNAME"] = "APERTURE"
-        aperture_hdu.header.set("NPIXSAP", None, "Number of pixels in optimal aperture")
-        aperture_hdu.header.set(
+        aperture_hdu_average.header["EXTNAME"] = "APERTURE"
+        aperture_hdu_average.header.set(
+            "NPIXSAP", None, "Number of pixels in optimal aperture"
+        )
+        aperture_hdu_average.header.set(
             "NPIXMISS", None, "Number of op. aperture pixels not collected"
         )
 
+        # Define extra FITS columns
+        cols = [
+            # Original FFI column of lower-left pixel in TPF.
+            fits.Column(
+                name="CORNER1",
+                format="I",
+                unit="pixel",
+                array=self.corner[:, 1],
+            ),
+            # Original FFI row of lower-left pixel in TPF.
+            fits.Column(
+                name="CORNER2",
+                format="I",
+                unit="pixel",
+                array=self.corner[:, 0],
+            ),
+            # 3D pixel quality mask
+            fits.Column(
+                name="PIXEL_QUALITY",
+                format=str(self.corr_flux[0].size) + "I",
+                dim=dims,
+                disp="B16.16",
+                array=self.pixel_quality,
+            ),
+            # Aperture as a function of time.
+            # Aperture has values 0 and 2, where 0/2 indicates the pixel is outside/inside the aperture.
+            # This format is used to be consistent with the aperture HDU from SPOC.
+            fits.Column(
+                name="APERTURE",
+                format=str(self.corr_flux[0].size) + "J",
+                dim=dims,
+                array=self.aperture_mask.astype("int32") * 2,
+            ),
+        ]
+
+        # Create table HDU for extra columns
+        table_hdu_extra = fits.BinTableHDU.from_columns(cols)
+        table_hdu_extra.header["EXTNAME"] = "EXTRAS"
+
         # Create hdulist and save to file
         self.hdulist = fits.HDUList(
-            [self.cube.output_primary_ext, table_hdu, aperture_hdu]
+            [
+                self.cube.output_primary_ext,
```

Comment:
Does this one have the updated header keywords, target name, creator, version, etc.?

Reply:
Not yet, I was going to leave that for a future PR. It's on my to do list!

Reply:
ok, that's fine